### DISCUSSION TITLE
Server Not Connecting/Showing Up

### DISCUSSION THREAD

**Discussion by NadulaG** [TGRHavoc/live_map-interface#56] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/51786507/110326277-6f36cc80-7fe6-11eb-9d22-916c4dc99f6f.png)
I am not able to see my server in the `Select a Server list`. In addition, when I try to press `Connect` in the control panel, it gets stuck on "Reconnecting". I am hosting the web interface on XAMPP with Apache. Here is my config.json: [https://pastebin.com/PqVrc2kh](https://pastebin.com/PqVrc2kh). My IP and FiveM port are filled in correctly.

**Comment by TGRHavoc** [comment]
If your website is using HTTPS then, you need to set up a reverse proxy for the blip URL and the socket port in order for it to work.
If you don't know how to do this, just make the website use HTTP.

**Comment by NadulaG** [comment]
Hmm.. my website is just running locally (127.0.0.1) on XAMPP. As far as I can tell, it is not using HTTPS.

Could this TypeError in the console have something to do with my issue?
![image](https://user-images.githubusercontent.com/51786507/110331681-6ac1e200-7fed-11eb-8435-d37b69022daa.png)

**Comment by TGRHavoc** [comment]
Could you load up the debug.html page to see what errors get thrown with that page?

**Comment by NadulaG** [comment]
Sure! When I go to `localhost/debug.html`, I get this:
![image](https://user-images.githubusercontent.com/51786507/110334028-50d5ce80-7ff0-11eb-88e5-0fd972eabd30.png)

(My FiveM server is being hosted on ZAP-Hosting by the way, not sure if that changes anything)

**Comment by TGRHavoc** [comment, marked as answer]
Oh yeah. Duh... You don't have a "map" array in the config.

Something like this should get it working for you (using the default map images)
```json
{
   "debug":true,
   "showIdentifiers":false,
   "tileDirectory":"images/tiles",
   "iconDirectory":"images/icons",
   "defaults":{
      "ip":"///",
      "fivemPort":"///",
      "socketPort":"30121",
      "liveMapName":"live_map-2.4.0"
   },
   "servers":{
      "G4 Roleplay":{}
   },
   "maps":[
      {
         "name":"Normal",
         "url":"{tileDirectory}/normal/minimap_sea_{y}_{x}.png",
         "minZoom":-2
      }
   ]
}
```

See: https://github.com/TGRHavoc/live_map-interface/blob/master/config.example.json

And if you wanted to use your own images, you would need the YTD files then, you can get a developer to run the extract_png.py file located https://github.com/TGRHavoc/live_map-interface/tree/master/images/tiles

**Comment by NadulaG** [comment]
Ah, gotcha. I completely overlooked that, haha. 

However, I am now getting these errors in the console:
![image](https://user-images.githubusercontent.com/51786507/110335257-a78fd800-7ff1-11eb-89f9-45e962c24afc.png)

It seems like it is not able to communicate with the FiveM server? This is the part of my server.cfg where the live map resource is started:
![image](https://user-images.githubusercontent.com/51786507/110335527-f76e9f00-7ff1-11eb-8225-bfeb0681dccf.png)

I have already generated blips on the server.

**Comment by TGRHavoc** [comment]
> My FiveM server is being hosted on ZAP-Hosting

Ah... I think there is some issues with lInux servers but, you can try the following:

1. Make sure the resource is actually starting on the server. Usually, setting the debug to `debug` or something should give you enough information to see if it's starting.,
2. If it is starting and listening, make sure it's allowed through the firewall.

**Comment by NadulaG** [comment]
I have set it to debug mode and it seems to be working. It is connected to the server and blips/players are displaying. So, what is the difference between this debug mode and the other modes? Would it be fine to just leave it like this?

**Comment by TGRHavoc** [comment]
There shouldn't be a difference that makes it work all of a sudden... Should just be the amount of messages being shown in console...

But yeah. Should be ok to keep the same but, it might get annoying if you have a lot of people connecting.

**Comment by NadulaG** [comment]
Alright, sounds good. Thanks so much for your help!
